lib/urlwatch/jobs.py

```diff
@@ -282,11 +285,21 @@ class BrowserJob(Job):
 
     __required__ = ('navigate',)
 
+    MAIN_THREAD = True
+    _session = None
+
     def get_location(self):
         return self.navigate
 
     def retrieve(self, job_state):
-        from requests_html import HTMLSession
-        session = HTMLSession()
-        response = session.get(self.navigate)
+        """
+        Retrieve web pages rendered in Chromium with JavaScript.
+        Must be called sequentially from the main thread
+        """
+        # all instances of BrowserJob share the same session
+        if not BrowserJob._session:
+            from requests_html import HTMLSession
+            BrowserJob._session = HTMLSession()
+        response = BrowserJob._session.get(self.navigate)
+        response.html.render()
```

Comment:
Is is actually important that the requests are run on the main thread, or is it just important that there are no requests happening at the same time? If it's the latter, maybe a simple [`threading.Lock`](https://docs.python.org/3/library/threading.html#lock-objects) would do the trick as well without needing to change `worker.py`?